### PR TITLE
Bump jenkins-test-harness to 2.56

### PR DESCRIPTION
**Pull request by slonopotamus**
This PR is expected to reveal bogus Windows tests that do not shutdown things properly. See jenkinsci/jenkins-test-harness#166

### Proposed changelog entries

None

### CODE REVIEW

**Comment by slonopotamus** [comment]
Err... Windows CI is still not working?

**Comment by oleg-nenashev** [comment]
@slonopotamus Yes, it is still disabled

**Comment by oleg-nenashev** [comment]
https://github.com/jenkinsci/jenkins/pull/4155 FTR

**Comment by slonopotamus** [comment]
Seems like Linux CI is also broken now.

**Comment by slonopotamus** [comment]
Do not merge this until Windows CI is working again because this PR is expected to break tests that worked otherwise.

File: pom.xml [hunk 1]
```diff
@@ -91,6 +91,7 @@ THE SOFTWARE.
     <project.patchManagement.url>https://api.github.com</project.patchManagement.url>
     <patch.tracker.serverId>jenkins-jira</patch.tracker.serverId>
 
+    <hamcrest.version>2.1</hamcrest.version>
     <matrix-auth.version>2.3</matrix-auth.version>
```
Comment (oleg-nenashev) [inline pom.xml]: Use dependencyManagement instead?

Comment (slonopotamus) [reply pom.xml]: jenkins-test-harness [doesn't do that| https://github.com/jenkinsci/jenkins-test-harness/blob/master/pom.xml#L110], why do this here differently?

Comment (oleg-nenashev) [reply pom.xml]: Less code, but no strong opinion

**Comment by timja** [comment]
Windows CI is re-enabled, re-triggering

**Comment by slonopotamus** [comment]
Nice. As I expected, some things failed. I plan to fix them within this PR.

**Comment by alecharp** [comment]
I added the label WIP based on your latest comment @slonopotamus. Thank you for noticing the problem.

**Comment by oleg-nenashev** [comment]
Ready to go @slonopotamus ?

**Review by oleg-nenashev — APPROVED** [review]
I approve it, because the issues seem to be fixed

**Comment by slonopotamus** [comment]
This PR only _seems_ to be fixed because Windows CI is disabled.

**Comment by slonopotamus** [comment]
JTH was already upgraded to 2.57 in #4392

**Pull request closed without merging**